SignInCheckIn/SignInCheckIn/Services/RoomService.cs

```diff
@@ -297,5 +297,66 @@ private void DeleteCurrentEventGroupsForRoomReservation(string authenticationTok
                 _eventRepository.DeleteEventGroups(authenticationToken, currentEventGroups.Select(g => g.Id));
             }
         }
+
+        public List<EventRoomDto> GetAvailableRooms(int roomId, int eventId)
+        {
+            var mpEvent = _eventRepository.GetEventById(eventId);
+
+            // exclude the origin room from the available rooms
+            var mpEventRooms = _roomRepository.GetRoomsForEvent(mpEvent.EventId, mpEvent.LocationId).Where(r => r.RoomId != roomId).ToList();
+
+            var eventRooms = Mapper.Map<List<MpEventRoomDto>, List<EventRoomDto>>(mpEventRooms);
+
+            var eventRoomIds = eventRooms.Select(r => r.EventRoomId).Distinct().ToList();
+            var bumpingRules = _roomRepository.GetBumpingRulesForEventRooms(eventRoomIds);
+
+            foreach (var rule in bumpingRules)
+            {
+                // set the rule id and priority on the matching event room - which is the "to" field, if it's a "bumping" event room
+                foreach (var room in eventRooms.Where(room => room.EventRoomId == rule.ToEventRoomId))
+                {
+                    room.BumpingRuleId = rule.BumpingRuleId;
+                    room.BumpingRulePriority = rule.PriorityOrder;
+                }
+            }
+
+            return eventRooms;
+        }
+
+        public List<EventRoomDto> UpdateAvailableRooms(int eventId, int roomId, List<EventRoomDto> eventRoomDtos)
```

Comment:
Should probably the user's authentication token as a parameter, and use it when calling the repository.